pkg/apis/resource/validation/validation.go

```diff
@@ -802,17 +802,24 @@ func validateNetworkDeviceData(networkDeviceData *resource.NetworkDeviceData, fl
 
 	allErrs = append(allErrs, validateSet(networkDeviceData.IPs, maxIPs,
```

Comment:
note , but I find this helpers very difficult to reason about

Reply:
ok, I got it now, the first function validats, the second function checsk for dupes